jax/experimental/jax2tf/tests/primitive_harness.py

```diff
@@ -585,3 +585,97 @@ def _fft_rng_factory(dtype):
           [arg, StaticArg(np.array([shift_amount], dtype=dtype))])
   for arg, dtype, shift_amount in shift_inputs
 )
+
+lax_select_and_gather_add = tuple(
+  Harness(f"shape={jtu.format_shape_dtype_string(shape, dtype)}_selectprim={select_prim}_windowdimensions={window_dimensions}_windowstrides={window_strides}_padding={padding}_basedilation={base_dilation}_windowdilation={window_dilation}",
+          lax._select_and_gather_add,
+          [RandArg(shape, dtype), RandArg(shape, dtype), StaticArg(select_prim),
+           StaticArg(window_dimensions), StaticArg(window_strides),
+           StaticArg(lax.padtype_to_pads(shape, window_dimensions, window_strides,
+                                         padding)),
+           StaticArg(base_dilation),
+           StaticArg(window_dilation)],
+          shape=shape,
+          dtype=dtype,
+          window_dimensions=window_dimensions,
+          window_strides=window_strides,
+          padding=padding,
+          base_dilation=base_dilation,
+          window_dilation=window_dilation)
+  for dtype in jtu.dtypes.all_floating
+  for shape in [(4, 6)]
+  for select_prim in [lax.le_p, lax.ge_p]
+  for window_dimensions in [(2, 1), (1, 2)]
+  for window_strides in [(1, 1), (2, 1), (1, 2)]
+  for padding in ['VALID', 'SAME']
+  for base_dilation in [(1, 1)]
+  for window_dilation in [(1, 1)]
+) + tuple(
+  Harness(f"explicitpadding_shape={jtu.format_shape_dtype_string(shape, dtype)}_selectprim={select_prim}_windowdimensions={window_dimensions}_windowstrides={window_strides}_padding={padding}_basedilation={base_dilation}_windowdilation={window_dilation}",
+          lax._select_and_gather_add,
+          [RandArg(shape, dtype), RandArg(shape, dtype), StaticArg(select_prim),
+           StaticArg(window_dimensions), StaticArg(window_strides),
+           StaticArg(padding), StaticArg(base_dilation),
+           StaticArg(window_dilation)],
+          shape=shape,
+          dtype=dtype,
+          window_dimensions=window_dimensions,
+          window_strides=window_strides,
+          padding=padding,
+          base_dilation=base_dilation,
+          window_dilation=window_dilation)
+  for dtype in jtu.dtypes.all_floating
+  for shape in [(4, 6)]
+  for select_prim in [lax.le_p, lax.ge_p]
+  for window_dimensions in [(2, 1), (1, 2)]
+  for window_strides in [(1, 1), (2, 1), (1, 2)]
+  for padding in [[(0, 3), (1, 2)]]
+  for base_dilation in [(1, 1)]
+  for window_dilation in [(1, 1)]
+) + tuple(
+  Harness(f"shape={jtu.format_shape_dtype_string(shape, dtype)}_selectprim={select_prim}_windowdimensions={window_dimensions}_windowstrides={window_strides}_padding={padding}_basedilation={base_dilation}_windowdilation={window_dilation}",
```

Comment:
It is not obvious to me that the name of the harness will not conflict with others using the same name pattern.

Reply:
Disambiguated 2d/4d tests so that names don't collide in `cb32206ee3`